akka-docs/rst/java/http/routing-dsl/directives/form-field-directives/formFields.rst

```diff
@@ -1,106 +0,0 @@
-.. _-formFields-java-:
-
-formFields
-==========
-
-Signature
----------
-
-::
-
-    def formFields(field: <FieldDef[T]>): Directive1[T]
-    def formFields(fields: <FieldDef[T_i]>*): Directive[T_0 :: ... T_i ... :: HNil]
-    def formFields(fields: <FieldDef[T_0]> :: ... <FieldDef[T_i]> ... :: HNil): Directive[T_0 :: ... T_i ... :: HNil]
-
-The signature shown is simplified and written in pseudo-syntax, the real signature uses magnets. [1]_ The type
-``<FieldDef>`` doesn't really exist but consists of the syntactic variants as shown in the description and the examples.
-
-.. [1] See `The Magnet Pattern`_ for an explanation of magnet-based overloading.
-.. _`The Magnet Pattern`: http://spray.io/blog/2012-12-13-the-magnet-pattern/
-
-Description
------------
-Extracts fields from requests generated by HTML forms (independently of ``HttpMethod`` used).
-
-Form fields can be either extracted as a String or can be converted to another type. The parameter name
-can be supplied either as a String or as a Symbol. Form field extraction can be modified to mark a field
-as required, optional, or repeated, or to filter requests where a form field has a certain value:
-
-``"color"``
-    extract value of field "color" as ``String``
-``"color".?``
-    extract optional value of field "color" as ``Option[String]``
-``"color" ? "red"``
-    extract optional value of field "color" as ``String`` with default value ``"red"``
```

Comment:
sure all of these exist in java? I had to add the `optional with default` version to parameter directives